server/komodo-server/src/main/java/org/komodo/openshift/TeiidOpenShiftClient.java

```diff
@@ -586,10 +581,8 @@ public void createDataSource(DefaultSyndesisDataSource scd) throws Exception {
         }
 
         //now that the komodoname is set, we can create the properties
-        Map<String, String> properties = scd.convertToDataSourceProperties();
-        properties.put(ID, scd.getId());
-
-        this.metadata.createDataSource(toUse, dsType, encryptionComponent.decrypt(properties));
+        TeiidDataSource teiidDS = scd.createDataSource(toUse);
```

Comment:
Maybe pass the DefaultSyndesisDataSource into a registerDataSource, then we can just get rid of the getDataSource check and handle everything in the DefaultMetadataInstance.

Reply:
corrected as suggested